Sources/NiceComponents/Config.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  Config.swift
+//  NiceComponents
+//
+//  Created by Brendan on 2021-01-29.
+//  Copyright © 2022 Steamclock Software. All rights reserved.
+//
+
+import os
+import SwiftUI
+
+/// Global settings for all components.
+/// Styles defined here will be applied to any components that don't define their own.
+public struct Config {
+    /// Your current component configuration.
+    /// Note that you can only set this configuration once, ideally during app startup. Subsequent updates will be ignored.
+    public static var current: Config {
+        get {
+            return _current
+        }
+
+        set {
+            guard !hasSetConfig else {
+                os_log("Error! Config has already been set once, at startup. Ignoring config update. ")
```

Comment:
Would suck to introduce a dependancy on Steamclog, or build some pluggable error reporting mechanism just for like a handful of errors, but it ALSO sucks that any errors reported via `os_log` are basically just going into a black hole in our existing apps. Not sure if there is any near term solution here, but we should maybe think about about if we want to try and get Steamclog supporting os_log in some way eventually.

Reply:
This is a great point - I'll file a separate ticket to look into this.

I also think there's an argument for this just being a fatalError as well - ideally this is something that gets caught in dev anyways

Reply:
This is something that has come up before for Steamclog, and I have done a bit of investigation on: https://github.com/steamclock/steamclog-swift/issues/89.